src/OpenLoco/src/Vehicles/OrderManager.cpp

```diff
@@ -98,6 +116,37 @@ namespace OpenLoco::Vehicles::OrderManager
         }
     }
 
+    void deleteOrder(VehicleHead* head, uint16_t orderOffset)
+    {
+        // Find out what type the selected order is
+        OrderRingView orders(head->orderTableOffset, orderOffset);
+        auto& selectedOrder = *(orders.begin());
+
+        // Bookkeeping: change order table size
+        // TODO: this should probably be done after shifting orders? Following original sub for now
+        auto removeOrderSize = kOrderSizes[enumValue(selectedOrder.getType())];
+        head->sizeOfOrderTable -= removeOrderSize;
+        numOrders() -= removeOrderSize;
+
+        // Are we removing an order that appears before the current order? Move back a bit
+        if (head->currentOrder > orderOffset)
+        {
+            head->currentOrder -= removeOrderSize;
+        }
+
+        // Ensure we don't move beyond the order table size
+        if (head->currentOrder + 1 >= head->sizeOfOrderTable)
+        {
+            head->currentOrder = 0;
+        }
```

Comment:
Either of these is still not enough. Removing all orders up to and including the current order leads to a crash. Not sure why yet, but I've had enough for today :-)